### PR TITLE
[Merged by Bors] - chore(SetTheory/Ordinal/FixedPoint): Rewrite dumb theorems about zero function

### PR DESCRIPTION
**Pull request by vihdzp**
Two theorems referred to the function `fun x => 0 * x`, which is just the zero function. Also, added applied versions of theorems referring to function equality.

---

The name `deriv_zero` is used for `deriv f 0 = nfp f 0` and `deriv 0 = id`. Should either of these be renamed?

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Ordinal/FixedPoint.lean [hunk 1]
```diff
@@ -542,26 +555,14 @@ theorem nfp_mul_zero (a : Ordinal) : nfp (a * ·) 0 = 0 := by
...
-
 theorem nfp_mul_eq_opow_omega {a b : Ordinal} (hb : 0 < b) (hba : b ≤ (a^omega)) :
     nfp (a * ·) b = (a^omega.{u}) := by
   rcases eq_zero_or_pos a with ha | ha
   · rw [ha, zero_opow omega_ne_zero] at hba ⊢
-    rw [Ordinal.le_zero.1 hba, nfp_zero_mul]
+    rw [Ordinal.le_zero.1 hba]
+    convert congr_fun nfp_zero 0
+    ext
+    rw [zero_mul]
```
Comment (YaelDillies) [inline Mathlib/SetTheory/Ordinal/FixedPoint.lean:565]:
```suggestion
    simp_rw [Ordinal.le_zero.1 hba, zero_mul, nfp_zero]
```
maybe?

File: Mathlib/SetTheory/Ordinal/FixedPoint.lean [hunk 2]
```diff
@@ -477,6 +477,19 @@ theorem deriv_eq_enumOrd (H : IsNormal f) : deriv f = enumOrd (Function.fixedPoi
 theorem deriv_eq_id_of_nfp_eq_id {f : Ordinal → Ordinal} (h : nfp f = id) : deriv f = id :=
   (IsNormal.eq_iff_zero_and_succ (deriv_isNormal _) IsNormal.refl).2 <| by simp [h]
 
+@[simp]
+theorem nfp_zero : nfp 0 = id := by
```
Comment (YaelDillies) [inline Mathlib/SetTheory/Ordinal/FixedPoint.lean:481]: Can we also have `nfp_zero_left : nfp 0 a = a`? This is the more useful lemma for rewriting

**Review by YaelDillies — APPROVED** [review]
Thanks! 🚀

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**